main.cpp

```diff
@@ -1 +1,33 @@
+#include <iostream> 
+#include <cstdlib>
+#include <iomanip>
+#include "coordinates.h"
+#include <vector> 
 
+using namespace std;
+
+int main() {
+    // loop that reads comma-delimited lines from standard input one line at a time and constructs a vector of coordinates 
+string coords;
+CoordVec x; 
+Coordinate userInput;
+
+// Using unix pipe to get coords for input  
+while (getline(cin, coords)){
+        // get whole line
+        int commaLocation = coords.find(",");
+        string lat = coords.substr(0, commaLocation -1); 
+        string longitude = coords.substr(commaLocation + 1, coords.back());
+        // cout << lat << longitude << endl; in correct order 
+        // push variable into a vector 
+        userInput.first = atof(lat.c_str());
```

Comment:
This doesn't account for the fact that each line starts with an identifier. It's not just latitude and longitude. That identifier needs to be stripped out. This is corrupting the coordinates; the output of the program is a coordinate that doesn't exist in the file.

Reply:
Because each line has two commas you want to look for the comma and create a substring twice.